File: frontend/optool/src/services/pm/sprint.ts
import { api } from 'src/boot/axios'

export type SprintStatus = 'PLANNED' | 'ACTIVE' | 'COMPLETED'

export type Sprint = {
  id: string
  projectId: string
  name: string
  goal: string | null
  status: SprintStatus
  startDate: string | null
  endDate: string | null
  issueCount: number
  createdAt: string
  updatedAt: string
}

export async function listSprints(projectId: string) {
  const { data } = await api.get<Sprint[]>(`/pm/projects/${projectId}/sprints`)
  return data
}

export async function createSprint(projectId: string, payload: {
  name: string
  goal?: string
  start_date?: string
  end_date?: string
}) {
  const { data } = await api.post<Sprint>(`/pm/projects/${projectId}/sprints`, payload)
  return data
}

export async function updateSprint(projectId: string, sprintId: string, payload: {
  name?: string
  goal?: string
  status?: SprintStatus
  start_date?: string | null
  end_date?: string | null
}) {
  const { data } = await api.patch<Sprint>(`/pm/projects/${projectId}/sprints/${sprintId}`, payload)
  return data
}

export async function deleteSprint(projectId: string, sprintId: string) {
  await api.delete(`/pm/projects/${projectId}/sprints/${sprintId}`)
}

File: app/models/pm/sprint.py
from __future__ import annotations

from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field


SprintStatus = Literal["PLANNED", "ACTIVE", "COMPLETED"]


class SprintCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    goal: Optional[str] = Field(None, max_length=500)
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None


class SprintPatch(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    goal: Optional[str] = Field(None, max_length=500)
    status: Optional[SprintStatus] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None


class SprintOut(BaseModel):
    id: str
    project_id: str
    name: str
    goal: Optional[str]
    status: SprintStatus
    start_date: Optional[datetime]
    end_date: Optional[datetime]
    issue_count: int = 0
    created_at: datetime
    updated_at: datetime
